src/main/java/org/graylog/integrations/pagerduty/PagerDutyNotification.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.pagerduty;
+
+import org.graylog.events.notifications.EventNotification;
+import org.graylog.events.notifications.EventNotificationContext;
+import org.graylog.events.notifications.EventNotificationException;
+import org.graylog.integrations.pagerduty.client.PagerDutyClient;
+import org.graylog.integrations.pagerduty.client.PagerDutyClient.PagerDutyClientException;
+import org.graylog.integrations.pagerduty.dto.PagerDutyResponse;
+
+import javax.inject.Inject;
+import java.util.List;
+
+/**
+ * Main class that focuses on event notifications that should be sent to PagerDuty.
+ *
+ * @author Edgar Molina
+ *
+ */
+public class PagerDutyNotification implements EventNotification
+{
+    private final PagerDutyClient pagerDutyClient;
+
+    @Inject
+    PagerDutyNotification(PagerDutyClient pagerDutyClient) {
+        this.pagerDutyClient = pagerDutyClient;
+    }
+
+    public interface Factory extends EventNotification.Factory {
+        @Override
+        PagerDutyNotification create();
+    }
+
+    @Override
+    public void execute(EventNotificationContext ctx) throws EventNotificationException {
+        try {
+            PagerDutyResponse response = pagerDutyClient.trigger(ctx);
+            List<String> errors = response.getErrors();
+            if (errors != null && errors.size() > 0) {
+                throw new IllegalStateException(
+                        "There was an error triggering the PagerDuty event, details: " + errors);
+            }
+        } catch (PagerDutyClientException e) {
+            throw new EventNotificationException (
+                    "There was an exception triggering the PagerDuty event.", e);
+        } catch (Exception e) {
+            throw new EventNotificationException (
```

Comment:
`final Notification systemNotification = notificationService.buildNow()
					.addNode(nodeId.toString())
					.addType(Notification.Type.GENERIC)
					.addSeverity(Notification.Severity.NORMAL)
					.addDetail("SlackEventNotification send error ", e.toString());

			notificationService.publishIfFirst(systemNotification);`

The above notification should be published here isn't ? we follow the pattern in `SlackPlugin` . how ever i donot know what is the purpose of this . Any thoughts on this.

Reply:
I'll be honest, I haven't done a deep dive on error handling in this plugin yet.  My thinking is that we should raise a notification on the Graylog UI only in situations where user action is required to make the plugin work (the same situations where we would want to throw a `PermanentEventNotificationException`) but not in situations where the problem might be transient.

So if the user has entered an invalid PagerDuty key, then we might want to throw a `PermanentEventNotificationException` and raise a notification on the UI.  But if there was a network timeout trying to do the POST to the PagerDuty API, then we might want to throw a `TemporaryEventNotificationException` and not raise a notification on the UI.

Looking at the HTTP Notification and Email Notification, the HTTP Notification never uses UI notifications while the Email Notification does.

Reply:
Fun fact I just learned: PagerDuty doesn't seem to do any validation of the `Routing Key`.  If the user puts in a bad routing key, they just won't get the notification in PagerDuty.

Reply:
so yeah repeating the question again, do we need `NotificationService` as a dependency in this class and do we call ,
`notificationService.publishIfFirst(systemNotification);` in this exception block ? probably @bernd can help with our queries.

Reply:
> Fun fact I just learned: PagerDuty doesn't seem to do any validation of the `Routing Key`. If the user puts in a bad routing key, they just won't get the notification in PagerDuty.

validation is not `critical` and can be addressed later. :)